File: 06. Sorting/Bubblesort.java
import java.io.*;
import java.util.*;
class BubbleSort{
    
   
    
    public static void main (String[] args) {
        Scanner sc=new Scanner(System.in);
        int n,j;
        System.out.println("Enter size of array");
        n=sc.nextInt();
        int a[]=new int[n];
        for(int i=0;i<n;i++)
            a[i]=sc.nextInt();
      
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n - i - 1; j++){
                if( arr[j] > arr[j + 1]){
                    
                    // swapping
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    
                }
            }
        }
        for(int i = 0; i < n; i++){
            System.out.print(a[i] + " ");
        }
    }
}

File: 06. Sorting/bubble_sort.c
#include<stdio.h>

void bubbleSort(int arr[], int n)
{
    int i,j,temp;
    for(i=0;i<n;i++)
    {
      for(j=0;j<n-1-i;j++)
      {
          if(arr[j]>arr[j+1])
          {
              temp = arr[j];
              arr[j] = arr[j+1];
              arr[j+1] = temp;
          }
      }
    }
}

int main()
{
    int n,i;

    printf("Enter the size of array:");
    scanf("%d",&n);

    int arr[n];
    printf("Enter %d elements:\n",n);
    for(i=0;i<n;i++)
        scanf("%d",&arr[i]);
    printf("\nUnsorted array:");
    for(i=0;i<n;i++)
        printf("%d ",arr[i]);

    bubbleSort(arr,n);
    printf("\nSorted array:");
    for(i=0;i<n;i++)
        printf("%d ",arr[i]);
    return 0;
}


